Demos/CBS/Allocator/CBSCostFunctions.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from Demos.CBS.Allocator.CBSAllocatorHelpers import HighLevelNode
+
+
+class CostFunction(ABC):
+    """
+    The cost function CBS should optimize
+    """
+    failed_allocation_valid = False  # Whether an allocation where not all agents have a path is valid
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def __call__(node: "HighLevelNode") -> float:
+        pass
+
+
+class PathLength(CostFunction):
+    """
+    Minimize the total number of allocated ticks
+    """
+    failed_allocation_valid = False
```

Comment:
will this always have a solution?

Reply:
CBS might not find a solution if failed_allocation_valid if False.
Is this what you mean?